spring-boot-autoconfigure/src/test/java/org/axonframework/springboot/PooledStreamingEventProcessorIntegrationTest.java

```diff
@@ -103,22 +160,50 @@ public Serializer eventSerializer() {
         }
 
         @Bean
-        public ConfigurerModule defaultToPooledStreaming() {
+        public ConfigurerModule configureEventProcessors(ListenerInvocationErrorHandler latchedErrorHandler) {
             return configurer -> configurer.eventProcessing()
                                            .usingPooledStreamingEventProcessors()
                                            .registerPooledStreamingEventProcessorConfiguration(
-                                                   (config, builder) -> builder.initialSegmentCount(1)
+                                                   "upcaster-test", (config, builder) -> builder.initialSegmentCount(1)
+                                           )
+                                           .registerListenerInvocationErrorHandler(
+                                                   "once-test", c -> latchedErrorHandler
+                                           )
+                                           .registerPooledStreamingEventProcessorConfiguration(
+                                                   "once-test", (config, builder) -> builder.initialSegmentCount(16)
                                            );
         }
 
         @Bean
+        @ConditionalOnProperty("upcaster-test")
         public MultiUpcaster multiUpcaster() {
             return new MultiUpcaster();
         }
 
         @Bean
-        public EventHandlingComponent eventHandlingComponent() {
-            return new EventHandlingComponent();
+        @ConditionalOnProperty("upcaster-test")
+        public UpcasterTestEventHandlingComponent upcasterTestEventHandlingComponent() {
+            return new UpcasterTestEventHandlingComponent();
+        }
+
+        @Bean
+        @ConditionalOnProperty("once-test")
+        public HandlingOnceEventHandlingComponent onceTestEventHandlingComponent() {
+            return new HandlingOnceEventHandlingComponent();
+        }
+
+        @Bean
+        public CountDownLatch errorCountLatch(@Value("${errorCount:2}") int errorCount) {
+            return new CountDownLatch(errorCount);
+        }
+
+        @Bean
+        public ListenerInvocationErrorHandler latchedErrorHandler(CountDownLatch errorCountLatch) {
+            return (exception, event, eventHandler) -> {
+                System.out.println("REACHEDREACHEDREACHEDREACHEDREACHEDREACHEDREACHED");
```

Comment:
I don't think we need this system.out here.

Reply:
Did you know you can add non-halting breakpoints in IntelliJ that just write a line to the console when the line is reached? At least you won't run the risk of getting these types of lines in production. I have some nice stories of this happening that I can share over a 🍺 😉.

Reply:
Aaah yes, I recall seeing something like that. I should use that option more, thanks for the reminder.
And uh, I'll hold you up on getting some drinks with stories attached ;-)